### PR TITLE
142 update

### PR DESCRIPTION
**Pull request by npanpaliya**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/main/CONTRIBUTING.md)?
- [ ] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/main/doc/)?
- [ ] Did you write any [tests](https://github.com/open-ce/open-ce/blob/main/tests/) to validate this change?  

## Description

Refreshed xgboost to 1.4.2..

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -20,64 +23,127 @@ requirements:
     - {{ compiler('cxx') }}
     - cmake {{ cmake }}
     - make
-    - git >={{ git }}
```
Comment (jayfurmanek) [inline recipe/meta.yaml:23]: Why remove these?

Comment (jayfurmanek) [reply recipe/meta.yaml:23]: oh..this is the metapkg..

File: recipe/meta.yaml [hunk 2]
```diff
@@ -57,22 +88,46 @@ outputs:
...
+        - cudatoolkit {{ cudatoolkit }}            #[build_type == 'cuda']
+        - nccl {{ nccl }}                          #[build_type == 'cuda']
       run:
         - cudatoolkit {{ cudatoolkit }}        #[build_type == 'cuda']
         - nccl {{ nccl }}                      #[build_type == 'cuda']
-        - _xgboost_select {{ xgboost_select_version }} 
         - libgcc-ng {{ libgcc }}
         - libstdcxx-ng {{ libstdcxx }}
+      run_constrained:
+        - lightgbm-proc * {{ build_type }}
```
Comment (bnemanich) [inline recipe/meta.yaml:104]: This should be xgboost, not lightgbm.

Comment (npanpaliya) [reply recipe/meta.yaml]: Thanks! Fixed this.

**Review by bnemanich — APPROVED** [review]

**Review by jayfurmanek — APPROVED** [review]

**Pull request merged**